### PR TITLE
docs: add info about using LLMs for coding

### PR DESCRIPTION
**Pull request by kjappelbaum**
## Summary by Sourcery

Introduce a new documentation section explaining how to use large language models for coding tasks

Documentation:
- Add a tutorial callout detailing LLM-assisted code generation, completion, debugging, and refactoring
- Outline limitations of LLM-generated code and emphasize the need for verification
- Provide best practices for crafting prompts and integrating LLM suggestions
- Include links to the OpenAI API reference and a prompt engineering guide

### CODE REVIEW

File: quarto_book/tutorial-coding.qmd [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+### Detailed Spec + Prototype
+```
+"Function signature:
+```python
+def analyze_spectrum(csv_path: str, method: str) -> dict:
+```
+Requirements: baseline correction (polynomial & asymmetric least squares), smoothing, peak detection
+Edge cases: noisy baseline, missing data, overlapping peaks
+First produce spec, then minimal prototype with synthetic data and 3 unit tests."
+```
```
Comment (MrtinoRG) [inline quarto_book/tutorial-coding.qmd:136]: I am not sure if here the formatting of the markdown will be the correct because of how the inverted quotes are

File: quarto_book/tutorial-coding.qmd [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+"Suggest 3 architectures for CLI batch-processing tool (dependencies, performance, ease of use). 
+Compare trade-offs, then implement the best option."
+```
+
+### Debug & Fix
+```
+"Error trace: [paste]
+Code: [paste]
+Expected: [behavior]
+Provide minimal fix + test to prevent regression."
```
Comment (MrtinoRG) [inline quarto_book/tutorial-coding.qmd:149]: You are very non verbose with the LLMs...

File: quarto_book/tutorial-coding.qmd [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+[Research found that many MCP environments store long-term API keys for third-party services in plaintext on the local filesystem, often with insecure, world-readable permissions.](https://blog.trailofbits.com/2025/04/30/insecure-credential-storage-plagues-mcp/) 
+[GitHub now provides secret scanning and push protection specifically for MCP workflows.](https://github.blog/changelog/2025-08-13-github-mcp-server-secret-scanning-push-protection-and-more/)
+
+### Prompt Injection Threats
+
+[AgentHopper (an AI virus) malware self-propagates through Git repositories - when AI agents get compromised, it embeds universal injection payloads that trigger when other developers pull infected code.](https://embracethered.com/blog/posts/2025/agenthopper-a-poc-ai-virus/)
+[Indirect injection attacks hide malicious instructions in web content that activate when LLMs summarize pages, bypassing traditional security measures because AI agents operate with broad authentication access.](https://brave.com/blog/comet-prompt-injection/)
```
Comment (MrtinoRG) [inline quarto_book/tutorial-coding.qmd:191]: I also like this one from Simon: https://simonwillison.net/2025/Jun/16/the-lethal-trifecta/

File: quarto_book/tutorial-coding.qmd [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+## Essential Practices
+
+- **Set clear specs** (signature, inputs/outputs, constraints, edge cases, style).
+- **Ask for options** (multiple approaches) and compare trade-offs.
+- **Work in small increments**: function → test → run → adjust.
+- **Use error feedback & tests to drive iteration**.
+- **Prototype early**: get something working soon, then refine.
+- **Use bots for review & improvement**: but don't assume perfection.
+- **Always human oversight**: review, test, verify.
+- **Use very detailed instructions when using an agent**: We often use one LLM to write the spec/instructions, then feed that into another or into an agent for implementation.
```
[the remark this answers is not in the document]
Comment (kjappelbaum) [reply quarto_book/tutorial-coding.qmd:56]: Do you have a suggestions on how to improve this text?

File: quarto_book/tutorial-coding.qmd (hunk 2 above)
Comment (kjappelbaum) [reply quarto_book/tutorial-coding.qmd:125]: would you suggest being less verbose?

File: quarto_book/tutorial-coding.qmd [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+---
+
+## Security Considerations
+
+**Critical Risks**: [Research shows](https://cybersecuritynews.com/security-nightmares-on-coding-using-llms/) [much AI-generated code contains security vulnerabilities](https://blog.himanshuanand.com/posts/2025-08-22-llm-vibe-coding-security-nightmare/). LLMs are trained on insecure code examples.
+
+
+### Common Vulnerabilities
+
+You can see some examples for issues [in](https://blog.vidocsecurity.com/blog/vibe-coding-security-vulnerabilities) [various](https://www.databricks.com/blog/passing-security-vibe-check-dangers-vibe-coding) [blog posts](https://hackaday.com/2025/04/12/vibe-check-false-packages-a-new-llm-security-risk/).
```
[the remark this answers is not in the document]
Comment (kjappelbaum) [reply quarto_book/tutorial-coding.qmd]: i revised this
